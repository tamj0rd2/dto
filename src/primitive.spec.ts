import { readdirSync } from 'fs'
import { resolve } from 'path'

const fixturePrefix = __filename.replace(`${__dirname}/`, '').replace('.spec.ts', '')
const files = readdirSync(resolve(process.cwd(), 'src', '__fixtures__', fixturePrefix))

files.forEach((fileName) => {
  test(fileName, () => {
    expect(`${fixturePrefix}/${fileName}`).toMatchDiagnosticsSnapshot()
  })
})
