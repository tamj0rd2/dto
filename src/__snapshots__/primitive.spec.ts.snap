// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`boolean.ts 1`] = `
Object {
  "errorCount": 4,
  "fileName": "primitive/boolean.ts",
  "messages": Array [
    "primitive/boolean.ts:7(14) Type 'null' is not assignable to type 'boolean'.",
    "primitive/boolean.ts:9(14) Type 'undefined' is not assignable to type 'boolean'.",
    "primitive/boolean.ts:11(14) Type 'string' is not assignable to type 'boolean'.",
    "primitive/boolean.ts:13(14) Type '{ bad: string; }' is not assignable to type 'boolean'.",
  ],
}
`;

exports[`null.ts 1`] = `
Object {
  "errorCount": 3,
  "fileName": "primitive/null.ts",
  "messages": Array [
    "primitive/null.ts:7(14) Type 'undefined' is not assignable to type 'null'.",
    "primitive/null.ts:9(14) Type '123' is not assignable to type 'null'.",
    "primitive/null.ts:11(14) Type '{ hey: string; }' is not assignable to type 'null'.",
  ],
}
`;

exports[`number.ts 1`] = `
Object {
  "errorCount": 4,
  "fileName": "primitive/number.ts",
  "messages": Array [
    "primitive/number.ts:7(14) Type 'null' is not assignable to type 'number'.",
    "primitive/number.ts:9(14) Type 'undefined' is not assignable to type 'number'.",
    "primitive/number.ts:11(14) Type 'string' is not assignable to type 'number'.",
    "primitive/number.ts:13(14) Type '{ bad: string; }' is not assignable to type 'number'.",
  ],
}
`;

exports[`optional-boolean.ts 1`] = `
Object {
  "errorCount": 3,
  "fileName": "primitive/optional-boolean.ts",
  "messages": Array [
    "primitive/optional-boolean.ts:9(14) Type 'undefined' is not assignable to type 'boolean | null'.",
    "primitive/optional-boolean.ts:11(14) Type '\\"123897\\"' is not assignable to type 'boolean | null'.",
    "primitive/optional-boolean.ts:13(14) Type '{ bad: string; }' is not assignable to type 'boolean | null'.
  Type '{ bad: string; }' is not assignable to type 'true'.",
  ],
}
`;

exports[`optional-null.ts 1`] = `
Object {
  "errorCount": 3,
  "fileName": "primitive/optional-null.ts",
  "messages": Array [
    "primitive/optional-null.ts:7(14) Type 'undefined' is not assignable to type 'null'.",
    "primitive/optional-null.ts:9(14) Type '123' is not assignable to type 'null'.",
    "primitive/optional-null.ts:11(14) Type '{ hey: string; }' is not assignable to type 'null'.",
  ],
}
`;

exports[`optional-number.ts 1`] = `
Object {
  "errorCount": 3,
  "fileName": "primitive/optional-number.ts",
  "messages": Array [
    "primitive/optional-number.ts:9(14) Type 'undefined' is not assignable to type 'number | null'.",
    "primitive/optional-number.ts:11(14) Type '\\"123897\\"' is not assignable to type 'number | null'.",
    "primitive/optional-number.ts:13(14) Type '{ bad: string; }' is not assignable to type 'number'.",
  ],
}
`;

exports[`optional-string.ts 1`] = `
Object {
  "errorCount": 4,
  "fileName": "primitive/optional-string.ts",
  "messages": Array [
    "primitive/optional-string.ts:9(14) Type '28947' is not assignable to type 'string | null'.",
    "primitive/optional-string.ts:11(14) Type 'undefined' is not assignable to type 'string | null'.",
    "primitive/optional-string.ts:13(14) Type 'false' is not assignable to type 'string | null'.",
    "primitive/optional-string.ts:15(14) Type '{ hello: string; }' is not assignable to type 'string'.",
  ],
}
`;

exports[`string.ts 1`] = `
Object {
  "errorCount": 5,
  "fileName": "primitive/string.ts",
  "messages": Array [
    "primitive/string.ts:7(14) Type 'number' is not assignable to type 'string'.",
    "primitive/string.ts:9(14) Type 'undefined' is not assignable to type 'string'.",
    "primitive/string.ts:11(14) Type 'boolean' is not assignable to type 'string'.",
    "primitive/string.ts:13(14) Type 'null' is not assignable to type 'string'.",
    "primitive/string.ts:15(14) Type '{ hello: string; }' is not assignable to type 'string'.",
  ],
}
`;
